internal/api/utils.go

```diff
@@ -265,56 +260,109 @@ func TransformComponentUnits(jsonData datatypes.JSON) map[string]interface{} {
 			if math.Abs(cpuInCores) < 1 {
 				cpuInMillicores := cpuInCores * 1000
 				cpu["amount"] = math.Round(cpuInMillicores) // millicore values are rounded & don't require decimal precision
-				cpu["format"] = "millicores"
+				cpu["format"] = "m"
 			} else {
 				cpu["amount"] = truncateToThreeDecimalPlaces(cpuInCores)
-				cpu["format"] = "cores"
+				cpu["format"] = nil
 			}
 		}
 		return nil
 	}
 
+	// Current
+	current_config, ok := data["current"].(map[string]interface{})
+	if !ok {
+		log.Error("current not found in JSON")
+	}
+
+	for _, section := range []string{"limits", "requests"} {
+
+		sectionObject, ok := current_config[section].(map[string]interface{})
+		if ok {
+			memory, ok := sectionObject["memory"].(map[string]interface{})
+			if ok {
+				err := convertMemory(memory)
+				if err != nil {
+					fmt.Printf("error converting memory in %s: %v\n", sectionObject, err)
+					continue
+				}
+			}
+			cpu, ok := sectionObject["cpu"].(map[string]interface{})
+			if ok {
+				err := convertCPU(cpu)
+				if err != nil {
+					fmt.Printf("error converting cpu in %s: %v\n", sectionObject, err)
+					continue
+				}
+			}
+		}
+	}
+
 	/*
 		Recommendation data is available for three periods
+		under cost and performance keys
 		For each of these actual values will be present in
 		below mentioned dataBlocks > request and limits
 	*/
 
-	for _, period := range []string{"long_term", "medium_term", "short_term"} {
-		intervalData, ok := durationBased[period].(map[string]interface{})
+	// Recommendations
+	recommendation_terms, ok := data["recommendation_terms"].(map[string]interface{})
+	if !ok {
+		log.Error("recommendation_terms not found in JSON")
+	}
+
+	for _, period := range []string{"short_term", "medium_term", "long_term"} {
+		intervalData, ok := recommendation_terms[period].(map[string]interface{})
 		if !ok {
 			continue
 		}
 
-		for _, dataBlock := range []string{"current", "config", "variation"} {
-			recommendationSection, ok := intervalData[dataBlock].(map[string]interface{})
-			if !ok {
-				continue
-			}
+		// Hack 
+		// remove nil equivalent monitoring_start_time in API response
+		monitoring_start_time := intervalData["monitoring_start_time"]
+		if monitoring_start_time == "0001-01-01T00:00:00Z" {
+			delete(intervalData, "monitoring_start_time")
+		}
```

Comment:
I wish to have this separated with more details on why we are implementing it. If its because kruize is not doing it and in future we should remove it then adding related JIRA can be helpful as well.

Reply:
Here as well? Do you think this deserve to be discussed in Kruize sync up?

Reply:
I don't think this need a sync with Kruize, I have updated the comment block